Refactor

bzgaspar opened the pull request:
Transfer Cart Item functions to Cart Modal. Separate component for quantity buttons to be reused in product details. Create ImageAndDetails that is used in product banner and product details page. Add enums to a separate file. Refactor Back button to Navigation button to be customize to back or forward.  Add semantic tags. Create See Product Button that is used in banners & product list page. Eliminate unnecessary styles.Remove redundant colors.  Refactor See product button. Refactor homepage.

[preview](https://audiophile-e-commerce-website-git-refactor-cartitem-itsmenek.vercel.app?_vercel_share=rd4pOx7TAMPQB8klpv27Uq2eHumnE2qW)

bzgaspar committed:
refactor: Refactor

Transfer Cart Item functions to Cart Modal. Separate component for quantity buttons to be reused in product details. Create ImageAndDetails that is used in product banner and product details page. Add enums to a separate file. Refactor Back button to Navigation button to be customize to back or forward.  Add semantic tags. Create See Product Button that is used in banners & product list page. Eliminate unnecessary styles.

Remove redundant colors.  Refactor See product button. Refactor homepage.

eremocal commented on src/components/Header/SubComponents/CartModal/CartModal.tsx:
```diff
@@ -33,6 +39,22 @@ export default function CartModal() {
+
+  const handleProductClick = (cartItem: CartItemType) => () => {
+    const product = cartItem.product
+    if (product.category && product.category.__typename !== 'Categories') return
```
What is the purpose of this? If there is a category, does this mean that when a user clicks on a product, it will do nothing?

eremocal commented on src/components/Shared/ImageAndDetail/ImageAndDetail.tsx:
```diff
@@ -0,0 +1,50 @@
+  }
+  return (
+    <>
+      <div className={Style['image-container']}>{renderImage()}</div>
```
what do you think if you move the condition here?

```
image &&
<div className={Style['image-container']}>{renderImage()}</div>
```

bzgaspar committed:
refactor:

Remove unnecessary condition in cart modal. Move the image condition in ImageAndDetail
